### PR TITLE
updating pref information on firefox support for web components features

### PR DESCRIPTION
**Pull request by chrisdavidmills**
I found out recently that Firefox prefs for web components features have changed. you now only need:

* dom.webcomponents.customelements.enabled for custom elements features.
* dom.webcomponents.shadowdom.enabled for shadow DOM features.

The old dom.webcomponents.enabled has now been completely removed.

I also added in support information for the "is" global attribute, seen as it is closely related.

### CODE REVIEW

**Comment by Elchi3** [comment]
The global `is`  attribute will get added in another PR already and I think the way the Firefox support is indicated there records the whole support history here. What do you think about it: https://github.com/mdn/browser-compat-data/pull/1282/files#diff-2a52c9b7d3948297f271ac5ef0a10070R546 ?

**Comment by chrisdavidmills** [comment]
That PR info looks good. I have therefore removed the "is" information from this one.

**Comment by Elchi3** [comment]
Over in the other PR the Firefox situation is described like this:

```json
   "firefox": [
              {
                "version_added": "59",
                "flags": [
                  {
                    "type": "preference",
                    "name": "dom.webcomponents.customelements.enabled",
                    "value_to_set": "true"
                  }
                ]
              },
              {
                "version_added": true,
                "version_removed": "59",
                "flags": [
                  {
                    "type": "preference",
                    "name": "dom.webcomponents.enabled",
                    "value_to_set": "true"
                  },
                  {
                    "type": "preference",
                    "name": "dom.webcomponents.customelements.enabled",
                    "value_to_set": "true"
                  }
                ]
              }
],
```
Does this describe the flag name changes better and would this apply to (all / some of?) the features you're changing as part of this PR?

**Comment by chrisdavidmills** [comment]
I guess this more accurate, strictly speaking. I'll update the others in this PR.

**Comment by Elchi3** [comment]
Looks good to me. Maybe @ExE-Boss wants to give this a review, too. 
And I think this PR is also conflicting with https://github.com/mdn/browser-compat-data/pull/1253

File: api/ShadowRoot.json [hunk 1]
```diff
@@ -21,36 +21,62 @@
...
+                {
+                  "type": "preference",
+                  "name": "dom.webcomponents.shadowdom.enabled",
+                  "value_to_set": "true"
+                }
+              ]
+            },
+            {
+              "version_added": true,
+              "version_removed": "59",
```
Comment (ExE-Boss) [inline api/ShadowRoot.json]: This feature was unavailable before Firefox 59.

File: api/ShadowRoot.json [hunk 2]
```diff
@@ -21,36 +21,62 @@
...
```
Comment (ExE-Boss) [inline api/ShadowRoot.json]: ditto

File: api/ShadowRoot.json [hunk 3]
```diff
@@ -94,36 +120,62 @@
...
+                  {
+                    "type": "preference",
+                    "name": "dom.webcomponents.shadowdom.enabled",
+                    "value_to_set": "true"
+                  }
+                ]
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
Comment (ExE-Boss) [inline api/ShadowRoot.json]: ditto

File: api/ShadowRoot.json [hunk 4]
```diff
@@ -94,36 +120,62 @@
...
```
Comment (ExE-Boss) [inline api/ShadowRoot.json]: ditto

File: api/ShadowRoot.json [hunk 5]
```diff
@@ -168,36 +220,62 @@
...
```
Comment (ExE-Boss) [inline api/ShadowRoot.json]: ditto

File: api/ShadowRoot.json [hunk 6]
```diff
@@ -168,36 +220,62 @@
...
```
Comment (ExE-Boss) [inline api/ShadowRoot.json]: ditto

File: api/ShadowRoot.json [hunk 7]
```diff
@@ -242,36 +320,62 @@
...
```
Comment (ExE-Boss) [inline api/ShadowRoot.json]: ditto

File: api/ShadowRoot.json [hunk 8]
```diff
@@ -242,36 +320,62 @@
...
```
Comment (ExE-Boss) [inline api/ShadowRoot.json]: ditto

File: css/selectors/slotted.json [hunk 9]
```diff
@@ -21,36 +21,62 @@
...
+                  {
+                    "type": "preference",
+                    "name": "dom.webcomponents.shadowdom.enabled",
+                    "value_to_set": "true"
+                  }
+                ]
+              },
+              {
+                "version_added": true,
+                "version_removed": "59",
```
Comment (ExE-Boss) [inline css/selectors/slotted.json]: ditto

File: css/selectors/slotted.json [hunk 10]
```diff
@@ -21,36 +21,62 @@
...
```
Comment (ExE-Boss) [inline css/selectors/slotted.json]: ditto

**Comment by chrisdavidmills** [comment]
OK, so for my latest trick, I've fixed slotted as per @ExE-Boss 's review. I've also reverted the ShadowRoot data back to what it currently is in master.

My thinking is that if we merge this PR, and then #1253 , all should work out correct? @Elchi3 ?

**Pull request merged**